llarp/util/time.hpp

```diff
@@ -29,7 +25,64 @@ namespace llarp
   struct time_delta
   {
     const TimePoint_t at;
+
+    /// how far away are we from the time point we have
+    template <typename delta_res_t = std::chrono::milliseconds>
+    auto
+    delta() const
+    {
+      return distance_from<delta_res_t>(DateClock_t::now());
+    }
+
+    template <typename delta_res_t = std::chrono::milliseconds>
+    auto
+    distance_from(const TimePoint_t& other) const
+    {
+      return std::chrono::duration_cast<delta_res_t>(other - at);
+    }
   };
```

Comment:
I don't understand why this is a struct; why not simply two free functions for this?

Reply:
I think this was a struct before because when using duration types for both timestamps and durations we couldn't tell which was which, and so had to wrap things that weren't timestamps into a `time_delta`.  That seems unnecessary now.

Reply:
indeed.